fontir/src/ir.rs

```diff
@@ -115,41 +116,122 @@ pub struct GlyphInstance {
-    pub contours: Vec<Contour>,
+    pub contours: Vec<BezPath>,
     /// List of glyph components.
     pub components: Vec<Component>,
 }
 
-/// A single glyph contour consisting of a list of points.
-pub type Contour = Vec<ContourPoint>;
-
-/// A single point in a glyph contour.
-#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
-pub struct ContourPoint {
-    pub x: f64,
-    pub y: f64,
-    pub typ: PointType,
-}
-
-/// Possible types of a point in a glyph contour, following UFO GLIF semantics.
-#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
-pub enum PointType {
-    Move,
-    Line,
-    OffCurve,
-    Curve,
-    QCurve,
-}
-
 /// A single glyph component, reference to another glyph.
 #[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
 pub struct Component {
     /// The name of the referenced glyph.
-    pub base: String,
+    pub base: GlyphName,
     /// Affine transformation to apply to the referenced glyph.
     pub transform: Affine,
 }
 
+/// Helps convert points-of-type to a bezier path.
+///
+/// Source formats tend to use streams of point-of-type. Curve manipulation is
+/// often easier on bezier path, so provide a mechanism to convert.
+#[derive(Debug)]
+pub struct GlyphPathBuilder<'a> {
+    glyph_name: &'a str,
+    offcurve: Vec<Point>,
+    path: BezPath,
+    last_move_to: Point,
+}
+
+impl<'a> GlyphPathBuilder<'a> {
+    pub fn new(glyph_name: &'a str) -> GlyphPathBuilder {
+        GlyphPathBuilder {
+            glyph_name,
+            offcurve: Vec::new(),
+            path: BezPath::new(),
+            // kurbo *requires* subpaths start with move so this isn't going to get used
+            last_move_to: Point::ZERO,
+        }
+    }
+
+    fn check_num_offcurve(&self, expected: impl Fn(usize) -> bool) -> Result<(), WorkError> {
+        if !expected(self.offcurve.len()) {
+            return Err(WorkError::InvalidSourceGlyph {
+                glyph_name: self.glyph_name.into(),
+                message: format!(
+                    "{} unexpected consecutive offcurve points {:?}",
+                    self.offcurve.len(),
+                    self.offcurve
+                ),
+            });
+        }
+        Ok(())
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.offcurve.is_empty() && self.path.elements().is_empty()
+    }
+
+    pub fn move_to(&mut self, p: impl Into<Point>) -> Result<(), WorkError> {
+        self.check_num_offcurve(|v| v == 0)?;
+        let p = p.into();
+        self.path.move_to(p);
+        self.last_move_to = p;
+        Ok(())
+    }
+
+    pub fn line_to(&mut self, p: impl Into<Point>) -> Result<(), WorkError> {
+        self.check_num_offcurve(|v| v == 0)?;
+        if self.is_empty() {
+            self.move_to(p)?;
+        } else {
+            self.path.line_to(p);
+        }
+        Ok(())
+    }
+
+    pub fn qcurve_to(&mut self, p: impl Into<Point>) -> Result<(), WorkError> {
+        todo!("Support qcurve to {}", p.into());
```

Comment:
Is there a good reason not to implement this now?

Reply:
Off the top of my head I wasn't sure what a qcurve was so I figured I'd leave it for a follow-on

Reply:
https://github.com/googlefonts/fontmake-rs/issues/110